pcli/src/command/tx.rs

```diff
@@ -215,73 +217,53 @@ impl TxCmd {
+                // use the default address.
+                let (claim_address, _dtk_d) =
+                    fvk.incoming().payment_address(source.unwrap_or(0).into());
+
+                let mut planner = Planner::new(OsRng);
+                planner.fee(swap_fee);
+                planner.swap(input, into, swap_claim_fee.clone(), claim_address)?;
+                let plan = planner
+                    .plan(app.view(), &fvk, source.map(Into::into))
+                    .await
+                    .context("can't plan swap transaction")?;
+
+                // Hold on to the swap plaintext to be able to claim.
+                let swap_plaintext = plan
                     .swap_plans()
                     .next()
-                    .expect("expected swap plan")
+                    .expect("swap plan must be present")
+                    .swap_plaintext
                     .clone();
 
-                // Submit the `Swap` transaction.
-                app.build_and_submit_transaction(swap_plan).await?;
-
-                // Wait for detection of the note commitment containing the Swap NFT.
-                let account_id = app.fvk.hash();
-                let note_commitment = swap_plan_inner.swap_body(&app.fvk).swap_nft.note_commitment;
-                // Find the swap NFT note associated with the swap plan.
-                let swap_nft_record = tokio::time::timeout(
-                    std::time::Duration::from_secs(20),
-                    app.view()
-                        .await_note_by_commitment(account_id, note_commitment),
-                )
-                .await
-                .context("timeout waiting to detect commitment of submitted transaction")?
-                .context("error while waiting for detection of submitted transaction")?;
-
-                // Now that the note commitment is detected, we can submit the `SwapClaim` transaction.
-                let swap_plaintext = swap_plan_inner.swap_plaintext;
-
-                // Fetch the batch swap output data associated with the block height
-                // and trading pair of the swap action.
-                //
-                // This batch swap output data comes from the client, it's necessary because
-                // the client has to encrypt the SwapPlaintext, however the validators *must*
-                // validate that the BatchSwapOutputData is correct when processing the SwapClaim!
-                let mut client = app.specific_client().await?;
-                let output_data: BatchSwapOutputData = client
-                    .batch_swap_output_data(BatchSwapOutputDataRequest {
-                        height: swap_nft_record.height_created,
-                        trading_pair: Some(swap_plaintext.trading_pair.into()),
-                    })
-                    .await?
-                    .into_inner()
-                    .try_into()
-                    .context("cannot parse batch swap output data")?;
```

Comment:
Note: all of this goes away, we no longer have to leak any queries to a fullnode to be able to produce the `SwapClaim`